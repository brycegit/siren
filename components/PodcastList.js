import React, { Component } from 'react';
import PodcastListCard from './PodcastListCard';
import { StyleSheet, Text, View, Button, TextInput, ScrollView, Image} from 'react-native';
import { Ionicons, FontAwesome } from '@expo/vector-icons';
import { connect } from 'react-redux';
import { actionCreators } from '../actions';
import Spinner from 'react-native-loading-spinner-overlay';

const mapStateToProps = (state) => ({
  podcasts: state.main.iTunesResult
})


class PodcastList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
      podcasts: props.podcasts || [],
      visible: false,
    }
  }

  getPodcasts () {
    query = this.state.text.slice().split().join('+');
    this.setState({text: "",  visible: true});
    fetch('http://itunes.apple.com/search?entity=podcast&term=' + query)
      .then(response => response.json())
      .then(response => {
        this.setState({
          podcasts: response.results,
          visible: false
        });
      })
      .catch(console.warn);
  }

  render() {
    return (
      <View style={styles.mainView}>
        <View style={styles.searchBar}>
          <TextInput underlineColorAndroid='rgba(0,0,0,0)' style={styles.searchInput} onChangeText={(text) => {this.setState({text});}} onSubmitEditing={this.getPodcasts.bind(this)} value={this.state.text}/>
          <Ionicons style={styles.searchButton} onPress={this.getPodcasts.bind(this)} size={30} color='grey' name="ios-search" />
        </View>
        <ScrollView style={styles.podcastList}>
          {this.state.visible?
            <Spinner visible={this.state.visible} textContent={"Searching..."} textStyle={{color: '#FFF'}} />  :
          this.state.podcasts.map((podcast, i) => (
              <PodcastListCard podcast={podcast} key={i}/>
            ))}
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  searchInput: {
    height: 40,
    width: 250,
    borderColor: 'gray',
    borderWidth: 1,
    padding: 10
  },
  mainView: {
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  searchButton: {
    marginLeft: 8,
  },
  podcastList:{
    width: '100%',
    marginBottom: 210,
  },
  searchBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 10,
  },
});

export default connect(mapStateToProps)(PodcastList);
