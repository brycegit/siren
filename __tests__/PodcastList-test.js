import React from 'react';
import PodcastList from '../components/PodcastList.js';
import * as renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../reducers/index';
const store = createStore(reducer);

var podcasts = [

  {"wrapperType":"track", "kind":"podcast", "artistId":127981066, "collectionId":561470997, "trackId":561470997, "artistName":"WNYC Studios", "collectionName":"Note to Self", "trackName":"Note to Self", "collectionCensoredName":"Note to Self", "trackCensoredName":"Note to Self", "artistViewUrl":"https://itunes.apple.com/us/artist/wnyc/id127981066?mt=2&uo=4", "collectionViewUrl":"https://itunes.apple.com/us/podcast/note-to-self/id561470997?mt=2&uo=4", "feedUrl":"http://feeds.wnyc.org/notetoself-podcast", "trackViewUrl":"https://itunes.apple.com/us/podcast/note-to-self/id561470997?mt=2&uo=4", "artworkUrl30":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/30x30bb.jpg", "artworkUrl60":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/60x60bb.jpg", "artworkUrl100":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/100x100bb.jpg", "collectionPrice":0.00, "trackPrice":0.00, "trackRentalPrice":0, "collectionHdPrice":0, "trackHdPrice":0, "trackHdRentalPrice":0, "releaseDate":"2017-04-05T04:00:00Z", "collectionExplicitness":"cleaned", "trackExplicitness":"cleaned", "trackCount":87, "country":"USA", "currency":"USD", "primaryGenreName":"Tech News", "contentAdvisoryRating":"Clean", "artworkUrl600":"http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/600x600bb.jpg", "genreIds":["1448", "26", "1318"], "genres":["Tech News", "Podcasts", "Technology"]},

  {"wrapperType":"track", "kind":"podcast", "collectionId":305727517, "trackId":305727517, "artistName":"Note to Beautiful Self", "collectionName":"Note to Beautiful Self", "trackName":"Note to Beautiful Self", "collectionCensoredName":"Note to Beautiful Self", "trackCensoredName":"Note to Beautiful Self", "collectionViewUrl":"https://itunes.apple.com/us/podcast/note-to-beautiful-self/id305727517?mt=2&uo=4", "feedUrl":"http://www.blogtalkradio.com/note2beautifulself/podcast", "trackViewUrl":"https://itunes.apple.com/us/podcast/note-to-beautiful-self/id305727517?mt=2&uo=4", "artworkUrl30":"http://is5.mzstatic.com/image/thumb/Music71/v4/41/27/e2/4127e299-472c-4fe6-c6bd-bb2594b39d03/source/30x30bb.jpg", "artworkUrl60":"http://is5.mzstatic.com/image/thumb/Music71/v4/41/27/e2/4127e299-472c-4fe6-c6bd-bb2594b39d03/source/60x60bb.jpg", "artworkUrl100":"http://is5.mzstatic.com/image/thumb/Music71/v4/41/27/e2/4127e299-472c-4fe6-c6bd-bb2594b39d03/source/100x100bb.jpg", "collectionPrice":0.00, "trackPrice":0.00, "trackRentalPrice":0, "collectionHdPrice":0, "trackHdPrice":0, "trackHdRentalPrice":0, "releaseDate":"2015-04-12T21:30:00Z", "collectionExplicitness":"cleaned", "trackExplicitness":"cleaned", "trackCount":39, "country":"USA", "currency":"USD", "primaryGenreName":"Careers", "contentAdvisoryRating":"Clean", "artworkUrl600":"http://is5.mzstatic.com/image/thumb/Music71/v4/41/27/e2/4127e299-472c-4fe6-c6bd-bb2594b39d03/source/600x600bb.jpg", "genreIds":["1410", "26", "1321"], "genres":["Careers", "Podcasts", "Business"]},

  {"wrapperType":"track", "kind":"podcast", "collectionId":1197015104, "trackId":1197015104, "artistName":"C David Baker", "collectionName":"Notes to Self", "trackName":"Notes to Self", "collectionCensoredName":"Notes to Self", "trackCensoredName":"Notes to Self", "collectionViewUrl":"https://itunes.apple.com/us/podcast/notes-to-self/id1197015104?mt=2&uo=4", "feedUrl":"http://cdbaker.podomatic.com/rss2.xml", "trackViewUrl":"https://itunes.apple.com/us/podcast/notes-to-self/id1197015104?mt=2&uo=4", "artworkUrl30":"http://is2.mzstatic.com/image/thumb/Music111/v4/8b/f6/0f/8bf60f3d-da43-c1d8-0d89-096f36a4b7e2/source/30x30bb.jpg", "artworkUrl60":"http://is2.mzstatic.com/image/thumb/Music111/v4/8b/f6/0f/8bf60f3d-da43-c1d8-0d89-096f36a4b7e2/source/60x60bb.jpg", "artworkUrl100":"http://is2.mzstatic.com/image/thumb/Music111/v4/8b/f6/0f/8bf60f3d-da43-c1d8-0d89-096f36a4b7e2/source/100x100bb.jpg", "collectionPrice":0.00, "trackPrice":0.00, "trackRentalPrice":0, "collectionHdPrice":0, "trackHdPrice":0, "trackHdRentalPrice":0, "releaseDate":"2017-04-07T20:41:00Z", "collectionExplicitness":"cleaned", "trackExplicitness":"cleaned", "trackCount":9, "country":"USA", "currency":"USD", "primaryGenreName":"Literature", "contentAdvisoryRating":"Clean", "artworkUrl600":"http://is2.mzstatic.com/image/thumb/Music111/v4/8b/f6/0f/8bf60f3d-da43-c1d8-0d89-096f36a4b7e2/source/600x600bb.jpg", "genreIds":["1401", "26", "1301"], "genres":["Literature", "Podcasts", "Arts"]}
];

//TESTING REACT COMPONENT
test('renders correctly without data', () => {
  const tree = renderer.create(
    //wrap component in provide so it has access to store
    <Provider store={store}>
      <PodcastList podcasts={podcasts} />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});



//To retake snapshot: npm test -- -u PodcastList
