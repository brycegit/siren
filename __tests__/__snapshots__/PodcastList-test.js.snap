// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "padding": 10,
          "width": 250,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <ScrollView
    style={
      Object {
        "marginBottom": 210,
        "width": "100%",
      }
    }
  >
    <View
      leftActionActivationDistance={200}
      leftActionReleaseAnimationConfig={null}
      leftActionReleaseAnimationFn={null}
      leftButtonWidth={75}
      leftButtonsActivationDistance={75}
      leftButtonsCloseReleaseAnimationConfig={null}
      leftButtonsCloseReleaseAnimationFn={null}
      leftButtonsOpenReleaseAnimationConfig={null}
      leftButtonsOpenReleaseAnimationFn={null}
      onLayout={[Function]}
      onLeftActionActivate={[Function]}
      onLeftActionComplete={[Function]}
      onLeftActionDeactivate={[Function]}
      onLeftActionRelease={[Function]}
      onLeftButtonsActivate={[Function]}
      onLeftButtonsCloseComplete={[Function]}
      onLeftButtonsCloseRelease={[Function]}
      onLeftButtonsDeactivate={[Function]}
      onLeftButtonsOpenComplete={[Function]}
      onLeftButtonsOpenRelease={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onPanAnimatedValueRef={[Function]}
      onRef={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onRightActionActivate={[Function]}
      onRightActionComplete={[Function]}
      onRightActionDeactivate={[Function]}
      onRightActionRelease={[Function]}
      onRightButtonsActivate={[Function]}
      onRightButtonsCloseComplete={[Function]}
      onRightButtonsCloseRelease={[Function]}
      onRightButtonsDeactivate={[Function]}
      onRightButtonsOpenComplete={[Function]}
      onRightButtonsOpenRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      onSwipeMove={[Function]}
      onSwipeRelease={[Function]}
      onSwipeStart={[Function]}
      rightActionActivationDistance={125}
      rightActionReleaseAnimationConfig={null}
      rightActionReleaseAnimationFn={null}
      rightButtonWidth={75}
      rightButtonsActivationDistance={75}
      rightButtonsCloseReleaseAnimationConfig={null}
      rightButtonsCloseReleaseAnimationFn={null}
      rightButtonsOpenReleaseAnimationConfig={null}
      rightButtonsOpenReleaseAnimationFn={null}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
      swipeReleaseAnimationConfig={
        Object {
          "duration": 250,
          "easing": [Function],
          "toValue": Object {
            "x": 0,
            "y": 0,
          },
        }
      }
      swipeReleaseAnimationFn={[Function]}
      swipeStartMinDistance={15}
    >
      <View
        style={
          Object {
            "marginLeft": -0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "justifyContent": "center",
                "paddingRight": 20,
              },
              Object {
                "backgroundColor": "#42f4c5",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Subscribe
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 2,
              "borderTopColor": "lightgrey",
              "borderTopWidth": 2,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.25,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={
                  Object {
                    "uri": "http://is3.mzstatic.com/image/thumb/Music111/v4/d6/04/42/d60442fb-dbc3-22e0-e23b-121cef5d511e/source/100x100bb.jpg",
                  }
                }
                style={
                  Object {
                    "height": 80,
                    "width": 80,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flex": 0.75,
                "height": 100,
                "justifyContent": "space-around",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              Note to Self
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 13,
                  "fontWeight": "400",
                  "marginBottom": 5,
                }
              }
            >
              WNYC Studios
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "backgroundColor": "#f4a442",
                    "padding": 2,
                  }
                }
              >
                 
                Tech News
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      leftActionActivationDistance={200}
      leftActionReleaseAnimationConfig={null}
      leftActionReleaseAnimationFn={null}
      leftButtonWidth={75}
      leftButtonsActivationDistance={75}
      leftButtonsCloseReleaseAnimationConfig={null}
      leftButtonsCloseReleaseAnimationFn={null}
      leftButtonsOpenReleaseAnimationConfig={null}
      leftButtonsOpenReleaseAnimationFn={null}
      onLayout={[Function]}
      onLeftActionActivate={[Function]}
      onLeftActionComplete={[Function]}
      onLeftActionDeactivate={[Function]}
      onLeftActionRelease={[Function]}
      onLeftButtonsActivate={[Function]}
      onLeftButtonsCloseComplete={[Function]}
      onLeftButtonsCloseRelease={[Function]}
      onLeftButtonsDeactivate={[Function]}
      onLeftButtonsOpenComplete={[Function]}
      onLeftButtonsOpenRelease={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onPanAnimatedValueRef={[Function]}
      onRef={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onRightActionActivate={[Function]}
      onRightActionComplete={[Function]}
      onRightActionDeactivate={[Function]}
      onRightActionRelease={[Function]}
      onRightButtonsActivate={[Function]}
      onRightButtonsCloseComplete={[Function]}
      onRightButtonsCloseRelease={[Function]}
      onRightButtonsDeactivate={[Function]}
      onRightButtonsOpenComplete={[Function]}
      onRightButtonsOpenRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      onSwipeMove={[Function]}
      onSwipeRelease={[Function]}
      onSwipeStart={[Function]}
      rightActionActivationDistance={125}
      rightActionReleaseAnimationConfig={null}
      rightActionReleaseAnimationFn={null}
      rightButtonWidth={75}
      rightButtonsActivationDistance={75}
      rightButtonsCloseReleaseAnimationConfig={null}
      rightButtonsCloseReleaseAnimationFn={null}
      rightButtonsOpenReleaseAnimationConfig={null}
      rightButtonsOpenReleaseAnimationFn={null}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
      swipeReleaseAnimationConfig={
        Object {
          "duration": 250,
          "easing": [Function],
          "toValue": Object {
            "x": 0,
            "y": 0,
          },
        }
      }
      swipeReleaseAnimationFn={[Function]}
      swipeStartMinDistance={15}
    >
      <View
        style={
          Object {
            "marginLeft": -0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "justifyContent": "center",
                "paddingRight": 20,
              },
              Object {
                "backgroundColor": "#42f4c5",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Subscribe
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 2,
              "borderTopColor": "lightgrey",
              "borderTopWidth": 2,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.25,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={
                  Object {
                    "uri": "http://is5.mzstatic.com/image/thumb/Music71/v4/41/27/e2/4127e299-472c-4fe6-c6bd-bb2594b39d03/source/100x100bb.jpg",
                  }
                }
                style={
                  Object {
                    "height": 80,
                    "width": 80,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flex": 0.75,
                "height": 100,
                "justifyContent": "space-around",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              Note to Beautiful Self
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 13,
                  "fontWeight": "400",
                  "marginBottom": 5,
                }
              }
            >
              Note to Beautiful Self
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "backgroundColor": "#f4a442",
                    "padding": 2,
                  }
                }
              >
                 
                Careers
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      leftActionActivationDistance={200}
      leftActionReleaseAnimationConfig={null}
      leftActionReleaseAnimationFn={null}
      leftButtonWidth={75}
      leftButtonsActivationDistance={75}
      leftButtonsCloseReleaseAnimationConfig={null}
      leftButtonsCloseReleaseAnimationFn={null}
      leftButtonsOpenReleaseAnimationConfig={null}
      leftButtonsOpenReleaseAnimationFn={null}
      onLayout={[Function]}
      onLeftActionActivate={[Function]}
      onLeftActionComplete={[Function]}
      onLeftActionDeactivate={[Function]}
      onLeftActionRelease={[Function]}
      onLeftButtonsActivate={[Function]}
      onLeftButtonsCloseComplete={[Function]}
      onLeftButtonsCloseRelease={[Function]}
      onLeftButtonsDeactivate={[Function]}
      onLeftButtonsOpenComplete={[Function]}
      onLeftButtonsOpenRelease={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onPanAnimatedValueRef={[Function]}
      onRef={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onRightActionActivate={[Function]}
      onRightActionComplete={[Function]}
      onRightActionDeactivate={[Function]}
      onRightActionRelease={[Function]}
      onRightButtonsActivate={[Function]}
      onRightButtonsCloseComplete={[Function]}
      onRightButtonsCloseRelease={[Function]}
      onRightButtonsDeactivate={[Function]}
      onRightButtonsOpenComplete={[Function]}
      onRightButtonsOpenRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      onSwipeMove={[Function]}
      onSwipeRelease={[Function]}
      onSwipeStart={[Function]}
      rightActionActivationDistance={125}
      rightActionReleaseAnimationConfig={null}
      rightActionReleaseAnimationFn={null}
      rightButtonWidth={75}
      rightButtonsActivationDistance={75}
      rightButtonsCloseReleaseAnimationConfig={null}
      rightButtonsCloseReleaseAnimationFn={null}
      rightButtonsOpenReleaseAnimationConfig={null}
      rightButtonsOpenReleaseAnimationFn={null}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          undefined,
        ]
      }
      swipeReleaseAnimationConfig={
        Object {
          "duration": 250,
          "easing": [Function],
          "toValue": Object {
            "x": 0,
            "y": 0,
          },
        }
      }
      swipeReleaseAnimationFn={[Function]}
      swipeStartMinDistance={15}
    >
      <View
        style={
          Object {
            "marginLeft": -0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "justifyContent": "center",
                "paddingRight": 20,
              },
              Object {
                "backgroundColor": "#42f4c5",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Subscribe
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "lightgrey",
              "borderBottomWidth": 2,
              "borderTopColor": "lightgrey",
              "borderTopWidth": 2,
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.25,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Image
                source={
                  Object {
                    "uri": "http://is2.mzstatic.com/image/thumb/Music111/v4/8b/f6/0f/8bf60f3d-da43-c1d8-0d89-096f36a4b7e2/source/100x100bb.jpg",
                  }
                }
                style={
                  Object {
                    "height": 80,
                    "width": 80,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "flex": 0.75,
                "height": 100,
                "justifyContent": "space-around",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingTop": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              Notes to Self
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 13,
                  "fontWeight": "400",
                  "marginBottom": 5,
                }
              }
            >
              C David Baker
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "backgroundColor": "#f4a442",
                    "padding": 2,
                  }
                }
              >
                 
                Literature
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
